src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -371,20 +399,8 @@ public final CountingReadFilter getMergedCountingReadFilter(final SAMFileHeader
         Utils.nonNull(samHeader);
         Utils.nonNull(aggregateFunction);
 
-        // start with the tool's default filters in the order they were specified, and remove any that were disabled
-        // on the command line
-        // if --disableToolDefaultReadFilters is specified, just initialize an empty list with initial capacity of user filters
-        final List<ReadFilter> finalFilters =
-                userArgs.getDisableToolDefaultReadFilters() ?
-                        new ArrayList<>(userArgs.getUserEnabledReadFilterNames().size()) :
-                        toolDefaultReadFilters.entrySet()
-                                .stream()
-                                .filter(e -> !isDisabledFilter(e.getKey()))
-                                .map(e -> e.getValue())
-                                .collect(Collectors.toList());
-
-        // now add in any additional filters enabled on the command line (preserving order)
-        final List<ReadFilter> clFilters = getAllInstances();
+        final List<ReadFilter> finalFilters = new ArrayList<>();
+        final List<ReadFilter> clFilters = getResolvedInstances();
         if (clFilters != null) {
             clFilters.stream()
                     .filter(f -> !finalFilters.contains(f)) // remove redundant filters
```

Comment:
Is `finalFilters` redundant here now? Seems like you always just end up with `finalFilters` equal to `clFilters` after this patch.

Reply:
You're right - redundancies are already removed, so this is all for nothing and can be simplified.